Source/stores.cpp

```diff
@@ -1600,16 +1701,15 @@ void S_WSellEnter()
  */
 void WitchRechargeItem()
 {
-	TakePlrsMoney(plr[myplr].HoldItem._iIvalue);
+	auto &myPlayer = plr[myplr];
+
+	TakePlrsMoney(myPlayer.HoldItem._iIvalue);
 
 	int idx = stextvhold + ((stextlhold - stextup) / 4);
 	storehold[idx]._iCharges = storehold[idx]._iMaxCharges;
 
-	int i = storehidx[idx];
-	if (i < 0)
-		plr[myplr].InvBody[INVLOC_HAND_LEFT]._iCharges = plr[myplr].InvBody[INVLOC_HAND_LEFT]._iMaxCharges;
-	else
-		plr[myplr].InvList[i]._iCharges = plr[myplr].InvList[i]._iMaxCharges;
+	int i = (storehidx[idx] >= 0) ? storehidx[idx] : static_cast<int>(INVLOC_HAND_LEFT);
+	myPlayer.InvList[i]._iCharges = myPlayer.InvList[i]._iMaxCharges;

```

Comment:
This doesn't look right to me. `INVLOC_HAND_LEFT` is now being used as the index into `InvList` instead of `InvBody`.

Reply:
Totally missed that, glad you spotted it :+1: